rosbag2_transport/src/rosbag2_transport/qos.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_TRANSPORT__QOS_HPP_
+#define ROSBAG2_TRANSPORT__QOS_HPP_
+
+#include "rclcpp/qos.hpp"
+
+#ifdef _WIN32
+// This is necessary because of a bug in yaml-cpp's cmake
+#define YAML_CPP_DLL
+// This is necessary because yaml-cpp does not always use dllimport/dllexport consistently
+# pragma warning(push)
+# pragma warning(disable:4251)
+# pragma warning(disable:4275)
+#endif
+#include "yaml-cpp/yaml.h"
+#ifdef _WIN32
+# pragma warning(pop)
+#endif
+
+
+namespace rosbag2_transport
+{
+/// Simple wrapper around rclcpp::QoS to provide a default constructor for YAML deserialization.
+class Rosbag2QoS : public rclcpp::QoS
+{
+public:
+  Rosbag2QoS()
+  : rclcpp::QoS(0) {}  // 0 history depth is always overwritten on deserializing
```

Comment:
I guess we could also default to `rclcpp::SystemsDefaultQos()` or something. But I think this doesn't really matter as far as I can understand your comment.

Reply:
Note that `rmw_qos_profile_system_default` is different than `rmw_qos_profile_default` - system_default is dependent on the rmw implementation, and because of that I don't think it's something anybody should ever use.

But, I will instantiate this with the default profile (not system_default)